app/Services/RestWrapper.js

```diff
@@ -36,7 +36,8 @@ class RestWrapper {
             .then((response) => {
                 return {
                     data: response().data,
-                    totalCount: response.headers()['x-total-count']
+                    totalCount: response().data.totalCount,
```

Comment:
Why do you ignore the header ?
You should check header first, then `totalCount`, then do a count of data elements.